js/jquery.mapael.js

```diff
@@ -222,216 +222,211 @@
              *
              * Update the current map
              * Refresh attributes and tooltips for areas and plots
-             * @param updatedOptions options to update for plots and areas
-             * @param newPlots new plots to add to the map
-             * @param deletedPlots plots to delete from the map (array, or "all" to remove all plots)
              * @param opt option for the refresh :
-             *  opt.animDuration animation duration in ms (default = 0)
-             *  opt.resetAreas true to reset previous areas options
-             *  opt.resetPlots true to reset previous plots options
-             *  opt.resetLinks true to reset previous links options
-             *  opt.afterUpdate Hook that allows to add custom processing on the map
+             *  opt.mapOptions: options to update for plots and areas
+             *  opt.replaceOptions: whether mapsOptions should entirely replace current map options, or just extend it
+             *  opt.opt.newPlots new plots to add to the map
              *  opt.newLinks new links to add to the map
-             *  opt.deletedLinks links to remove from the map (array, or "all" to remove all links)
+             *  opt.deletePlotKeys plots to delete from the map (array, or "all" to remove all plots)
+             *  opt.deleteLinkKeys links to remove from the map (array, or "all" to remove all links)
              *  opt.setLegendElemsState the state of legend elements to be set : show (default) or hide
+             *  opt.animDuration animation duration in ms (default = 0)
+             *  opt.afterUpdate Hook that allows to add custom processing on the map
              */
-            $self.on("update", function(e, updatedOptions, newPlots, deletedPlots, opt) {
-                var i = 0
-                    , animDuration = 0
-                    , elemOptions = {}
-                    // This function remove an element using animation (or not, depending on animDuration)
-                    // Used for deletedPlots and deletedLinks
-                    , fnRemoveElement = function(elem) {
-                    	// Unset all event handlers
-                    	Mapael.unsetHover(elem.mapElem, elem.textElem);
-                        if (animDuration > 0) {
-                            elem.mapElem.animate({"opacity":0}, animDuration, "linear", function() {
+            $self.on("update", function(e, opt) {
+                if (typeof opt === "object") {
```

Comment:
What do you think about replacing the huge condition `if (typeof opt === "object") { ...}` by `if (typeof opt !== "object") return;` in order to decrease the indentation level ?